CRM/Contribute/BAO/Contribution.php

```diff
@@ -2438,9 +2438,14 @@ public function composeMessageArray(&$input, &$ids, &$values, $recur = FALSE, $r
     if (empty($this->_component)) {
       $this->_component = CRM_Utils_Array::value('component', $input);
     }
+    if (isset($input['is_email_receipt'])) {
+      $values['is_email_receipt'] = $input['is_email_receipt'];
+    }
 
+    // Store values in a temp variable, so they don't get lost after _gatherMessageValues
+    $tmp_values = $values;
     //not really sure what params might be passed in but lets merge em into values
-    $values = array_merge($this->_gatherMessageValues($input, $values, $ids), $values);
+    $values = array_merge($this->_gatherMessageValues($input, $values, $ids), $tmp_values);
     if (!empty($input['receipt_date'])) {
       $values['receipt_date'] = $input['receipt_date'];
     }
```

Comment:
Yes we can, but if any extra options are passed through in `$values` that differ from any options after `_gatherMessageValues()`, they'll be lost anyway. 

e.g. If `receipt_from_email` is passed along in `$values`, but differs from whatever is retrieved after `_gatherMessageValues()`. 

Not sure if/when this would ever be a problem. If you think it's not an issue, let me know and we'll adjust the code.